python/lsst/dax/distribution/DataGenConnection.py

```diff
@@ -418,6 +429,47 @@ def clientRecvChunks(self):
             print("WARN clientRecvChunks problem with", msg, msg_chunks)
         return msg_chunks, problem
 
+    def clientReportTiming(self, timing_dict):
+        """Send a small dictionary of string keys and float time values.
+
+        Parameters
+        ----------
+        timing_dict : TimingDict
+            TimingDict object to send.
+
+        """
+        print("clientReportTiming C_TIMDCT", timing_dict)
+        time_msg = ''
+        c_sep = self.COMPLEXSEP
+        time_msg += str(timing_dict.count)
+        for tkey, tval in timing_dict.times.items():
+            time_msg += c_sep + str(tkey) + c_sep + str(tval)
+        self._send_msg(self.C_TIMDCT, time_msg)
+        return
+
+    def servRecvTiming(self):
+        """Receive the dictionary of timing information from the client
+
+        Return
+        ------
+        timing_dict : TimingDict
+            TimingDict object
+        """
+        print("servRecvTiming C_TIMDCT")
+        msg_id, msg, msg_len = self._recv_msg()
+        if msg_id != self.C_TIMDCT:
+            self.warnings += 1
+            raise DataGenError(f'ERROR servRecvTiming {str(msg_id)}:{str(msg_len)} {msg}')
```

Comment:
Why the warnings increment if it's not being returned in the raise()?

Reply:
At the end of running tests, warnings should be a specific number. The exception may have been caught and gone un-noticed.